### PR TITLE
Recommend users to directly specify dependencies when using them directly

### PR DESCRIPTION
**Pull request by johnlinp**
The discussion is in
https://lists.apache.org/thread.html/10b53804b5a18565f7f1a0677300d7f5c5fce539c94d9bad44efa6b1@%3Cusers.maven.apache.org%3E

### CODE REVIEW

File: content/apt/guides/introduction/introduction-to-dependency-mechanism.apt [hunk 1]
```diff
@@ -102,6 +102,14 @@ Introduction to the Dependency Mechanism
...
 
+ Although transitive dependencies can help you to include desired dependencies, it is a good practice that you directly
+ specify the dependencies you are directly using in your own project. This is true especially when the dependencies of
+ your project constantly changes their dependencies. Assume that your project A specifies a dependency on another
+ project B, and project B specifies a dependency on project C. If you are directly using components in project C, and
+ you don't specify project C in your project A, it may cause build failure when project B suddenly updates/removes its
+ dependency on project C. Another reason to directly specify dependencies is that it provides better documentation for
+ your project: one can learn more information by just reading the POM file in your project.
+
  []
```
Comment (hboutemy) [inline content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: your paragraph should go after the "[]" that mark the end of the bullet points

when writing documentation in site, you can easily check the rendering of your changes with "mvn site:run" then open your browser on localhost:8080

File: content/apt/guides/introduction/introduction-to-dependency-mechanism.apt [hunk 2]
```diff
@@ -102,6 +102,14 @@ Introduction to the Dependency Mechanism
      explicitly add a dependency on Z, at her option.  (It may be helpful to think of optional
      dependencies as "excluded by default.")
 
+ Although transitive dependencies can help you to include desired dependencies, it is a good practice that you directly
+ specify the dependencies you are directly using in your own project. This is true especially when the dependencies of
```
Comment (hboutemy) [inline content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: project -> source code

Comment (hboutemy) [inline content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: "This is true" -> "This best practice proves its value"

**Comment by hboutemy** [comment]
good improvement (and really appreciate to transform your findings when learning Maven into PRs :) )

perhaps adding a link to dependency:analyze mojo https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html could be useful, to show that there is some tooling available to ease this best practice

File: content/apt/guides/introduction/introduction-to-dependency-mechanism.apt (hunk 1 above)
Comment (johnlinp) [reply content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: I see. Thanks for the tips!

File: content/apt/guides/introduction/introduction-to-dependency-mechanism.apt (hunk 2 above)
Comment (johnlinp) [reply content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: I'll change that.

Comment (johnlinp) [reply content/apt/guides/introduction/introduction-to-dependency-mechanism.apt]: I'll fix it.

**Comment by johnlinp** [comment]
Hi @hboutemy,

Thanks for the review! I updated the paragraph with your suggestions. Please take a look at it.

**Comment by hboutemy** [comment]
merged
thank you

**Pull request closed without merging**